actions/google-docs/models/update_operations.py

```diff
@@ -0,0 +1,628 @@
+        #   * index +=text.length in each cell
+        #   *, index +=2 for the end of the table
+
+        # Reference on index calculation for tables:
+        # https://gist.github.com/tanaikech/3b5ac06747c8771f70afd3496278b04b?permalink_comment_id=4397902#gistcomment-4397902
+
+        # We added the table so the index shifted by one
+        current_index = start_index + 1
+
+        requests = []
+        style_requests = []
+
+        for row in rows:
+            current_index += 1
+            for cell in row:
+                current_index += 2
+                operations = cell.get_update_operations(current_index)
+                try:
+                    while True:
+                        op = next(operations)
+                        if isinstance(op, InsertTextRequest):
+                            requests.append(op)
+                        else:
+                            style_requests.append(op)
+                except StopIteration as exc:
+                    current_index = exc.value
+                    pass
+
+        # We first insert the text and then we apply the style of the text
+        yield from requests
+        yield from style_requests
+
+        current_index += 2
+
+        return current_index
+
+
+class BatchUpdateBody(_BaseUpdateRequest):
+    requests: list[UpdateRequest]
+
+    @classmethod
+    def from_raw_string(cls, text: str):
+        """Used to generate the body for a new document"""
+        return cls(requests=[InsertTextRequest.new(text, index=1)])
+
+    @classmethod
+    def from_markdown(cls, text: str):
+        text = text.strip()
+        current_index = 1
+        requests = []
+        images = []
+        for line in cls._get_lines(text):
+            operations = line.get_update_operations(start_index=current_index)
+            while True:
+                try:
+                    op = next(operations)
+                    if isinstance(op, InsertInlineImageRequest):
+                        # Adding an image breaks the calculated offsets, so we add them last.
+                        images.append(op)
+                    else:
+                        requests.append(op)
+                except StopIteration as exc:
+                    current_index = exc.value
+                    break
+
+        requests += images
+
+        for i, request in enumerate(requests):
+            print(f"([{i}]", request)
+
+        return cls(requests=requests)
+
+    def get_body(self):
+        return self.model_dump(
+            mode="json",
+            by_alias=True,
+            exclude_none=True,
+        )
+
+    @staticmethod
+    def _get_lines(text: str) -> Iterable[_Line | _TableLine]:
+        list_context = _ListContext()
+        table_context = _TableContext()
+
+        # Split by new lines
+        for new_line in text.split("\n"):
+            # Split by double spaces (new lines in Markdown)
+
+            for markdown_line in new_line.split("  "):
```

Comment:
Maybe it might be better to do a split based on regex exp.
Reviewing this, it doesn't look like 2 spaces.